guard-examples/cloudformation-resources/best-practices.guard

```diff
@@ -0,0 +1,67 @@
+# Auto-generated names are a CloudFormation best practice.
+# Resource Providers can use generateResourceIdentifier from IdentifierUtils (or equivalent)
+# This does not fail (yet?) for places where ${Resource}Name is a required property
+rule ensure_name_is_optional when required !empty {
+    required[*] {
+        this != 'Name'
+    }
+}
+
+# ReplacementStrategy should not be delete_then_create, except with very good reasons.
+# This breaks customers expectations of how Rollbacks work
+rule ensure_default_replacementStrategy when replacementStrategy exists {
+    replacementStrategy == 'create_then_delete'
+}
+
+# URLs should use https
+rule ensure_sourceUrl_uses_https when sourceUrl exists {
+    sourceUrl == /^https:/
+}
+
+# URLs should use https
+rule ensure_documentationUrl_uses_https when documentationUrl exists {
+    documentationUrl == /^https:/
+}
+
+# If a Tags property is defined, it should follow the same configuration as other resources
+# See https://github.com/aws-cloudformation/cloudformation-cli/blob/master/examples/schema/aws.common.types.v1.json
+# The common types are more strict than what we currently find in the schemas
+rule ensure_consistent_tags when properties.Tags exists {
```

Comment:
Not all AWS resources support tags, some (like BucketPolicy) don't because they attach to a resource that contain tags, some resources don't support tags at all, or don't support them at launch (or at least that used to happen).